### PR TITLE
[IMP] project_todo: remove reminder type activity from activity type

### PR DESCRIPTION
**Pull request by pash-odoo**
Before this commit, if we add todo from activity menu then it create a activity
with reminder type which is redundant with todo.
Remove reminder type activity from activity type and make to do activity type as
default when creating a todo from activity menu.

task-3196864

### CODE REVIEW

**Review by auon-odoo — APPROVED** [review]
Lgtm, thanks!

**Comment by xavierbol** [comment]
It seems you have some conflicts, can you resolve the conflicts when you have time? Thanks in advance! 🙂

**Comment by ppr-odoo** [comment]
> It seems you have some conflicts, can you resolve the conflicts when you have time? Thanks in advance! 🙂

@pash-odoo on holiday so i resolved the conflict and re-target upgrade branch.thanks

File: addons/project_todo/data/mail_activity_type_data.xml [hunk 1]
```diff
@@ -1,15 +0,0 @@
...
-        <record id="mail_activity_data_reminder" model="mail.activity.type">
-            <field name="name">Reminder</field>
-            <field name="category">reminder</field>
-            <field name="icon">fa-tasks</field>
-            <field name="delay_count">0</field>
-            <field name="sequence">20</field>
-        </record>
-        <record id="mail.mail_activity_data_todo" model="mail.activity.type">
-            <field name="category">reminder</field>
-        </record>
```
Comment (xavierbol) [inline addons/project_todo/data/mail_activity_type_data.xml:13]: I don't think it was intended to remove that.

File: addons/mail/models/mail_activity_type.py [hunk 2]
```diff
@@ -125,3 +125,13 @@ def _inverse_triggered_next_type_id(self):
                 activity_type.chaining_type = 'trigger'
             else:
                 activity_type.chaining_type = 'suggest'
+
+    @api.ondelete(at_uninstall=False)
+    def _unlink_except_todo(self):
+        if self.env.ref('mail.mail_activity_data_todo') in self:
+            raise UserError("The 'to-do' activity type is used to create reminders from the top bar menu and the command palette. Consequently, it cannot be archived or deleted.")
```
Comment (xavierbol) [inline addons/mail/models/mail_activity_type.py:132]: It should translated.
```suggestion
            raise UserError(_("The 'To-Do' activity type is used to create reminders from the top bar menu and the command palette. Consequently, it cannot be archived or deleted."))
```

File: addons/mail/models/mail_activity_type.py [hunk 3]
```diff
@@ -125,3 +125,13 @@ def _inverse_triggered_next_type_id(self):
...
+
+    def action_archive(self):
+        if self.env.ref('mail.mail_activity_data_todo') in self:
+            raise UserError("The 'to-do' activity type is used to create reminders from the top bar menu and the command palette. Consequently, it cannot be archived or deleted.")
```
Comment (xavierbol) [inline addons/mail/models/mail_activity_type.py:136]: It should be translated. 🙂 
```suggestion
            raise UserError(_("The 'To-Do' activity type is used to create reminders from the top bar menu and the command palette. Consequently, it cannot be archived or deleted."))
```

**Review by xavierbol — COMMENTED** [review]
Thanks for your work, I left 2 comments and after that it will look good to me. 🙂

**Review by xavierbol — APPROVED** [review]
LGTM, thanks for your work. 🙂 

robodoo r+

**Comment by ppr-odoo** [comment]
@xavierbol upgrade branch is ready. thanks

**Comment by xavierbol** [comment]
robodoo r+

**Pull request closed without merging**